### PR TITLE
Use the correct define (NET6_0_OR_GREATER)

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change

This PR takes the learnings of the defines from the net7 branch and uses the correct defines. This reduces the diff as well as makes sure that it is correct.

### CODE REVIEW

File: src/TestUtils/src/DeviceTests/AssertionExtensions.Android.cs [hunk 1]
```diff
@@ -33,7 +33,8 @@ void OnLayout(object? sender = null, AView.LayoutChangeEventArgs? e = null)
 			{
 				var view = (AView)sender!;
 
-				view.LayoutChange -= OnLayout;
+				if (view.Handle != IntPtr.Zero)
```
Comment (Eilon) [inline src/TestUtils/src/DeviceTests/AssertionExtensions.Android.cs:36]: Just checking, is this change actually related?

**Review by Eilon — APPROVED** [review]
Looks great! Just had 1 questiron.

File: src/TestUtils/src/DeviceTests/AssertionExtensions.Android.cs (hunk 1 above)
Comment (mattleibow) [reply src/TestUtils/src/DeviceTests/AssertionExtensions.Android.cs:36]: Not directly, but due to the async nature of this area of the tests it fails 34.8% of the time (I just made that up) So instead of a separate pr to fix that I am just adding it here. I think even this PR failed on that. And it is mostly safe due to it being test code.
